modules/c++/mt/include/mt/CPUAffinityThreadInitializerLinux.h

```diff
@@ -0,0 +1,75 @@
+/* =========================================================================
+ * This file is part of mt-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2019, MDA Information Systems LLC
+ *
+ * mt-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+#ifndef __MT_CPU_AFFINITY_THREAD_INITIALIZER_LINUX_H__
+#define __MT_CPU_AFFINITY_THREAD_INITIALIZER_LINUX_H__
+
+#if !defined(__APPLE_CC__)
+#if defined(__linux) || defined(__linux__)
+
+#include <memory>
+#include <sched.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#define gettid() syscall(SYS_gettid)
```

Comment:
This `#define` can go in the `.cpp` file to not pollute callers that include the header right?  And it looks like `gettid()`  is only called in one spot in the cpp, so do we need the `#define` at all vs. just calling this directly in the one spot?

Reply:
A relic of a bygone era...